engine/class_modules/monk/sc_monk.cpp

```diff
@@ -5440,16 +5471,20 @@ struct invoke_xuen_the_white_tiger_buff_t : public monk_buff_t<buff_t>
 
     for ( auto target : p->sim->target_non_sleeping_list )
     {
-      if ( p->get_target_data( target )->debuff.empowered_tiger_lightning->up() )
+      if ( p->find_target_data( target ) )
       {
-        double value = p->get_target_data( target )->debuff.empowered_tiger_lightning->value();
-        p->get_target_data( target )->debuff.empowered_tiger_lightning->current_value = 0;
-        if ( value > 0 )
+        auto td = p->get_target_data( target );
```

Comment:
This seems unnecessary. Why not use the result of find_target_data above? Same issue in the code below.

Reply:
The reason was that we can't (or shouldn't be able to) reset the current value with a const target data.
I know it looks a bit strange, but if that really is what the code should do I think it is the correct way.

Reply:
Oh right, failed to look at what's actually being done to the target data. Makes sense.